// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

//package;
syntax = "proto3";
package lib;

// Interface exported by the server.
service Client {
  rpc Ping(MessageBool) returns (MessageBool) {}
  rpc Command(GeneralMsg) returns (stream GeneralMsg) {}
  rpc DownloadFileCommand(GeneralMsg) returns (stream CFileMsg) {}
  rpc UploadFileCommand(stream CFileMsg) returns (GeneralMsg) {}
}
service AI {
  rpc Ping(MessageBool) returns (MessageBool) {}
  rpc Command(GeneralMsg) returns (stream GeneralMsg) {}
  rpc DownloadFileCommand(GeneralMsg) returns (stream CFileMsg) {}
  rpc UploadFileCommand(stream CFileMsg) returns (GeneralMsg) {}
}

service NetShell {
  rpc log(GeneralMsg) returns (GeneralMsg) {}
}

service Test {
  rpc Ping(MessageBool) returns (MessageBool) {}
}

// ====================================
// ========= for sending data =========
// ====================================
message MessageBool {
  bool value = 1;
}
message GeneralMsg {
  string id = 1;
  string msg = 2;
}
message CFileMsg {
  string id = 1;
  string file_info = 2;
  bytes data = 3;
}



